File: day06/golang/part2/part2.go
package main

import (
	"aoc/lib/jfile"
	"fmt"
)

const SIZE = 14

func uniq(text string) bool {
	d := make(map[rune]struct{})
	for _, c := range text {
		_, found := d[c]
		if found {
			return false
		}
		// else
		d[c] = struct{}{}
	}
	return true
}

func process(text string) int {
	for i := 0; i < len(text)-SIZE+1; i++ {
		sub := text[i : i+SIZE]
		if uniq(sub) {
			return i + SIZE
		}
	}
	return -1
}

func main() {
	text, _ := jfile.Read("input.txt")
	result := process(text)
	fmt.Println(result)
}

File: day06/python/part2.py
#!/usr/bin/env python3

import helper

SIZE = 14


def process(text: str) -> int:
    for i in range(len(text) - SIZE + 1):
        sub = text[i : i + SIZE]
        if len(sub) == len(set(sub)):
            return i + SIZE
        #
    #
    return -1


def main():
    text = helper.read("input.txt")
    result = process(text)
    print(result)


##############################################################################

if __name__ == "__main__":
    main()
